### ISSUE TITLE
"no-floating-promises" does not report failure with `.catch` / `.then(..., catch)`

### ISSUE THREAD

**Issue by vaskevich** [palantir/tslint#3936]
### Bug Report

- __TSLint version__: 5.9.1
- __TypeScript version__: 2.8.3
- __Running TSLint via__: (pick one) CLI

#### TypeScript code being linted

```ts
    new Promise(resolve => resolve()).catch(e => {throw e;});
```

with `tslint.json` configuration:

```json
{
  // ...
  "rules": {
    // ...
    "no-floating-promises": [true, "Bluebird"],
  }
}
```

#### Actual behavior
No error is reported.

#### Expected behavior
Would expect to be able to catch a floating promise via this rule, even if it requires a specific configuration parameter. This is because it's possible to throw errors in catch blocks, and not returning a floating promise in a chain can result in other bugs.

**Comment by vaskevich** [comment]
More context: basically would expect to be able to make the change in #3048 configurable.

**Comment by JoshuaKGoldberg** [comment]
☠️ *TSLint's time has come!* ☠️ 

TSLint is no longer accepting most feature requests per #4534. See **[typescript-eslint.io](https://typescript-eslint.io)** for the new, shiny way to lint your TypeScript code with **ESLint**. ✨ 

It was a pleasure open sourcing with you all!

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈 _(#4534)_  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
